src/Http/Routing/src/Builder/EndpointRoutingApplicationBuilderExtensions.cs

```diff
@@ -54,6 +55,10 @@ public static IApplicationBuilder UseRouting(this IApplicationBuilder builder)
             builder.Properties[EndpointRouteBuilder] = endpointRouteBuilder;
         }
 
+        // Add UseRouting function to properties so that middleware that can't reference UseRouting directly can call UseRouting via this property
+        // This is part of the global endpoint route builder concept
+        builder.Properties.TryAdd(UseRoutingKey, (object)UseRouting);
```

Comment:
I hate that this almost seems elegant in comparison to what we were doing before 😅